doc/design/zarr-support-3.md

```diff
@@ -0,0 +1,149 @@
+# Zarr Support
+A zarr archive contains directories and zarr files. 
+Zarr archives are stored in their entirety in S3.
+Zarr archives are represented by a single Asset through the API.
+A tree hash is computed as the zarr archive is uploaded.
+The tree hash details (all nodes) are stored on S3 to avoid bloating the DB.
+The intended use of Zarr archives is primarily to support use cases where data are too big to be copied or to be stored in a single file due to available storage limits. However, Zarr archives can be small as well.
+
+# Requirements
+1. Zarr archives are stored in a "directory" in S3.
+1. Each zarr archive corresponds to a single Asset.
+1. The CLI uses some kind of tree hashing scheme to compute a checksum for the entire zarr archive.
+1. The API verifies the checksum immediately after upload, so the validation status is available in the response to the upload REST request.
+1. The system can theoretically handle zarr files with ~1 million subfiles, each of size 64 * 64 * 64 bytes ~= 262 kilobytes.
+1. Zarr metadata must be mutable for zarr files in draft dandisets.
+1. Zarr archives must not be copied, as they are too large to reasonably store multiple copies.
+1. Zarr archives are immutable as long as they are contained by a published dandiset.
+
+# Implementation
+
+## API endpoints
+* **POST /api/zarr/**
+
+  Create a new zarr archive.
+
+  Returns a zarr ID
```

Comment:
may be then

```suggestion
  Create a new zarr archive and mint a UUID for it which would become `zarr_id`

  Returns the `zarr_id`
```
following [above](https://github.com/dandi/dandi-api/pull/574/files#r758787229)